File: app/src/main/java/com/zest/android/category/OnCategoryFragmentInteractionListener.java
package com.zest.android.category;

import com.zest.android.data.Category;

import java.util.List;

/**
 * To make an interaction between {@link CategoryFragment} and {@link CategoryViewModel}
 * <p>
 * Created by ZARA on 09/30/2018.
 */
interface OnCategoryFragmentInteractionListener {

    void showSubCategories(Category category);

    void setResult(List<Category> categories);
}

File: app/src/main/java/com/zest/android/category/OnCategoryFragmentInteractionListener.kt
package com.zest.android.category

import com.zest.android.data.Category

/**
 * To make an interaction between {@link CategoryFragment} and {@link CategoryViewModel}
 *
 * Created by ZARA on 09/30/2018.
 */
interface OnCategoryFragmentInteractionListener {

    fun showSubCategories(category: Category)

    fun setResult(categories: List<Category>)
}